rpi-update

```diff
@@ -233,70 +240,60 @@ function update_sdk {
 	fi
 }
 
-# Check if the bootloader is older than the latest critical release. An old
-# bootloader shouldn't block rpi-update so just inform the user that bootloader
-# is out of date.
-function check_eeprom_version {
-	local CURRENT_VERSION=""
-	local FIRST_VERSION=1557513636
-	# MIN VERSION for Sep 10 2019 EEPROM
-	local MIN_VERSION=1568112110
-	local HAVE_BOOTLOADER_EEPROM=0
-
-	# Skip EEPROM check if vcgencmd is missing because it won't be possible to
-	# check the version.
-	if ! command -v vcgencmd > /dev/null; then
-		return
-	fi
-
-	rev=
-	if [ -f /proc/cpuinfo ]; then
-		rev="$(sed -n '/^Revision/s/^.*: \(.*\)/\1/p' < /proc/cpuinfo)"
-	fi
-
-	if [ -n $rev ] && [ $(((0x$rev >> 23) & 1)) -ne 0 ] && [ $(((0x$rev >> 12) & 15)) -eq 3 ]; then
-		HAVE_BOOTLOADER_EEPROM=1
-	fi
+function download_bootloader_tools {
+	for tool in rpi-eeprom-update rpi-eeprom-config rpi-eeprom-digest; do
+		temp="$(mktemp)"
+		if ! eval curl -fs ${CURL_OPTIONS} "${BOOTLOADER_REPO_CONTENT_URI}/${BOOTLOADER_REV}/${tool}" --output "${temp}"; then
+			echo " !!! Failed to download ${tool}"
+			rm -rf "${temp}"
+			exit 1
+		fi
+		mv "${temp}" "/usr/bin/${tool}"
+		chmod a+rx "/usr/bin/${tool}"
+	done
+}
 
-	if [ "${HAVE_BOOTLOADER_EEPROM}" != 1 ]; then
-		return
+function download_bootloader_images {
+	bcm_chip=$1
+	latest_eeprom=$(eval curl -fs ${CURL_OPTIONS} ${BOOTLOADER_REPO_API_URI}/git/trees/master?recursive=1 | grep "firmware-${bcm_chip}/latest/pieeprom" | sed 's/.*\(pieeprom.*\)".*/\1/' | sort -r | head -n1)
+	if [[ -n "${latest_eeprom}" ]]; then
+		fw_path="/lib/firmware/raspberrypi/bootloader-${bcm_chip}/latest"
+		if [[ ! -d "${fw_path}" ]]; then
+			echo "Path ${fw_path} not found. Please install the rpi-eeprom package first"
+			return
```

Comment:
`return` or `exit 1`? This sounds fatal to me.

Reply:
Yes, let's make that fatal. We shouldn't use this unless the user has migrated the new firmware directory structure i.e. not Bullseye